system-contracts/contracts/L2MessageRootStorage.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.24;
+
+import {SystemContractBase} from "./abstract/SystemContractBase.sol";
+
+event MessageRootAdded(uint256 indexed chainId, uint256 indexed blockNumber, bytes32[] sides);
+error SidesLengthNotOne();
+error MessageRootAlreadyExists();
+
+/**
+ * @author Matter Labs
+ * @custom:security-contact security@matterlabs.dev
+ * @notice MessageRootStorage contract responsible for storing the message roots of other chains on the L2.
+ * @dev
+ */
+contract L2MessageRootStorage is SystemContractBase {
+    mapping(uint256 chainId => mapping(uint256 blockOrBatchNumber => bytes32 msgRoot)) public msgRoots;
```

Comment:
nit: add doc-comment for this field, even though it is self descriptive, the auditors would typically ask for it anyway